storage/innobase/handler/ha_innodb.cc

```diff
@@ -17896,6 +17894,35 @@ static bool innobase_get_index_column_cardinality(
   return (failure);
 }
 
+static bool innobase_is_tablespace_keyring_v1_encrypted(
+    const dd::Tablespace &tablespace, int &error) {
+  error = 0;
+  space_id_t id = 0;
```

Comment:
is this used only during upgrade?
If so, can you add assert ut_ad(srv_is_upgrade_mode);

Reply:
Not only for upgrade. I am checking this unconditionally for undo and system tablespaces.

Reply:
ok